tests/tier2/tc_2013_validate_unreachable_proxy_by_config.py

```diff
@@ -19,68 +20,89 @@ def test_run(self):
             results.setdefault('step1', []).append(res1)
             results.setdefault('step1', []).append(res2)
             results.setdefault('step1', []).append(res3)
             self.vw_option_del(option, filename=sysconfig_file)
 
         logger.info(">>>step2: run with bad proxy server and no_proxy")
-        for type, option in sorted(types.items(), key=lambda item:item[0]):
-            logger.info(">>%s: ------ %s test------" % (type, option))
-            logger.info("> run virt-who with bad {0}".format(option))
+        for type, option in sorted(types.items(), key=lambda item: item[0]):
+            logger.info("=== {0}: bad {1} test ===".format(type, option))
+            logger.info("+++ run virt-who with bad {0} +++".format(option))
             if option == "http_proxy":
                 value = "http://{0}".format(wrong_squid_server)
             if option == "https_proxy":
                 value = "https://{0}".format(wrong_squid_server)
             self.vw_option_add(option, value, filename=sysconfig_file)
             data, tty_output, rhsm_output = self.vw_start(exp_send=0, exp_error=True)
-            error_msg = "Connection refused|Cannot connect to proxy|Connection timed out"
+            error_msg = ["Connection refused|"
+                         "Cannot connect to proxy|"
+                         "Connection timed out"]
             res1 = self.op_normal_value(data, exp_error='1|2', exp_thread=1, exp_send=0)
-            res2 = self.vw_msg_search(rhsm_output, error_msg)
+            res2 = self.msg_validation(rhsm_output, error_msg)
             results.setdefault('step2', []).append(res1)
             results.setdefault('step2', []).append(res2)
-            logger.info("> configure no_proxy=[register_server] in /etc/sysconfig/virt-who")
+
+            logger.info("+++ configure no_proxy=[register_server] "
+                        "in /etc/sysconfig/virt-who +++")
             self.vw_option_add("no_proxy", register_server, sysconfig_file)
-            if hypervisor_type in ('libvirt-remote', 'hyperv', 'kubevirt'):
-                logger.info("virt-who connect hypervisor '{0}' not by proxy".format(hypervisor_type))
+            if (
+                    (
+                            "RHEL-8" in compose_id and hypervisor_type in (
+                            'libvirt-remote', 'hyperv', 'kubevirt')
+                    ) or
+                    (
+                            "RHEL-7" in compose_id and (
+                            ('http_proxy' in option and hypervisor_type in (
+                            'esx', 'libvirt-remote', 'xen', 'rhevm', 'kubevirt'))
+                            or
+                            ('https_proxy' in option and hypervisor_type in (
+                            'libvirt-remote', 'hyperv', 'kubevirt')))
+                    )
+            ):
+                '''virt-who connect hypervisor not by proxy'''
```

Comment:
The if statement is very complicated, hoping we can have some more test method for this scenarios.

Reply:
Simplified test steps and succeeded to debug locally.
```
=================== test session starts =====================
tests/tier2/tc_2013_validate_unreachable_proxy_by_config.py .                                                                              [100%]

=================== 1 passed in 369.38 seconds =================
```